src/shogun/evaluation/CrossValidation.cpp

```diff
@@ -264,32 +264,57 @@ float64_t CCrossValidation::evaluate_one_run()
 		m_machine->set_store_model_features(true);
 
 		/* do actual cross-validation */
+		#pragma omp parallel for if (get_global_parallel()->get_num_threads()!=1)
 		for (index_t i=0; i <num_subsets; ++i)
 		{
+			CMachine* machine;
+			CFeatures* features;
+			CLabels* labels;
+
+			if (get_global_parallel()->get_num_threads()==1)
+			{
+				machine=m_machine;
+			}
+			else
+			{
+				machine=(CMachine*)m_machine->clone();
```

Comment:
You *do* need them. Just because it worked doesnt mean it will always work. Basically, behaviour is undefined if the thing is not refed and you pass it around. Example

```
bar=foo->clone(); // ref-counter is 0
method_call(bar) // assume this method increases the counter once at the beginning, and then decreases afterwards -- it becomes 0 and the object is deleted
bar->method() // crash
```